Mathlib/Algebra/Algebra/Subalgebra/Basic.lean

```diff
@@ -371,7 +371,14 @@ instance : Module R S :=
 instance [Semiring R'] [SMul R' R] [Module R' A] [IsScalarTower R' R A] : IsScalarTower R' R S :=
   inferInstanceAs (IsScalarTower R' R (toSubmodule S))
 
-instance algebra' [CommSemiring R'] [SMul R' R] [Algebra R' A] [IsScalarTower R' R A] :
+/- More general form of `Subalgebra.algebra`.
+
+This instance should have low priority since it is slow to fail:
+before failing, it will cause a search through all `SMul K' K` instances,
+which can quickly get expensive.
+-/
+instance (priority := low) algebra' [CommSemiring R'] [SMul R' R] [Algebra R' A]
```

Comment:
How low is the priority `low`? IMHO it shouldn't be too low here, because it is already specific to `Subalgebra R A`, so it should have higher priority than instances that are completely general. Would a priority of 300 be low enough that more specific instances would be tried first?

Reply:
Also, does switching the order (i.e., using
```
[CommSemiring R'] [Algebra R' A] [SMul R' R]
```
make it faster to fail?

Reply:
Can we get away with them not being instances? I'll test once I get into my office.

Reply:
> How low is the priority `low`?
According to `Init.Notation`: `low = 100`, `mid = 500`, `default = 1000` and `high = 10000`.

> Would a priority of 300 be low enough that more specific instances would be tried first?

Any priority between 0 and 999 doesn't seem to change the results of the `IntermediateField` test case. So maybe `priority := mid` (i.e. 500)? Or we could even declare a new macro for priority, so we can write `priority := generalizedScalarInstancePrio` for all three of these instances of the same shape. (Better name suggestions very much welcome!)

Reply:
Wow. But how come this instance is even tried at some point, since there is no `Subalgebra` here? Does it try to fill in the typeclasses before realizing that the stuff it's looking at is not of the form `Subalgebra ...`?

Reply:
I'd prefer an explicit number for the priority, because then it's easier to see if you're above or below, and adjust things as needed.

Reply:
> But how come this instance is even tried at some point, since there is no `Subalgebra` here?

Coercing `E` to a sort gives `@Subtype L fun x => x ∈ E`, and the `∈` operator comes from the following:
```
instance : SetLike (IntermediateField K L) L :=
  ⟨fun S => S.toSubalgebra.carrier, by
    rintro ⟨⟨⟩⟩ ⟨⟨⟩⟩
    simp ⟩
```
So unfolding effectively casts `E` to a subalgebra and apply succeeds!

Reply:
> Can we get away with them not being instances? I'll test once I get into my office. 

`Subalgebra.algebra'` no but I didn't try terribly hard. `IntermediateField.algebra'` yes. 

I don't like having instances laying around that serve no purpose. But if people feel strongly about keeping, I won't fight.